docs/developer_guide.md

```diff
@@ -3,6 +3,55 @@
 Each sub-project has its own developer guide. If you would like to work across
 projects, these instructions should help you get started:
 
+
+### Install Dependencies
+
+Install shortfin dependencies
+```bash
+sudo apt update && sudo apt install -y clang lld
+```
+
+### Prepare your python environment
+
+Install:
+
+```
+python-is-python3 python3-venv python3-dev
+```
```

Comment:
Prefix these with `sudo apt install`

Also nit: set the syntax for code blocks (\`\`\`bash)